Tests/MapboxMapsTests/Style/StyleTransitionTests.swift

```diff
@@ -0,0 +1,40 @@
+import XCTest
+
+#if canImport(MapboxMaps)
+@testable import MapboxMaps
+#else
+@testable import MapboxMapsStyle
+#endif
+
+//swiftlint:disable explicit_top_level_acl explicit_acl
+class StyleTransitionTests: XCTestCase {
+
+    func testDecodeHasCorrectConversion() {
+        let jsonString = """
+        {
+            "duration": 1000,
+            "delay": 500
+        }
+        """
+
+        if let jsonData = jsonString.data(using: .utf8) {
+            let decodedTransition = try! JSONDecoder().decode(StyleTransition.self, from: jsonData)
+            let comparingTransition = StyleTransition(duration: 1.0, delay: 0.5)
+
+            XCTAssertEqual(comparingTransition.duration, decodedTransition.duration, "Duration should be equal to 1.0")
+            XCTAssertNotEqual(decodedTransition.duration, 1000, "Duration should be equal to 1.0")
+            XCTAssertEqual(comparingTransition.delay, decodedTransition.delay, "Delay should be equal to 0.5")
+            XCTAssertNotEqual(decodedTransition.delay, 500, "Duration should be equal to 1.0")
+        }
```

Comment:
These assertions are more complicated than necessary. Just have 1 assertion for each property that compares to a constant.